lib/i18n/tasks/data/tree/traversal.rb

```diff
@@ -143,18 +141,6 @@ def get_nodes_by_key_filter(root: false, &block)
         matches
       end
 
-      # @return [Siblings]
-      def intersect_keys(other_tree, key_opts = {}, &block)
```

Comment:
`intersect_keys` method is only called by test, so I removed it.

Reply:
This method is public and may be used by plugins / user extensions so let's keep it

Reply:
Thank you for reviewing. I overlooked the plugin system.

I restored the method and spec in 40c91279da5277079deba6bb1992189b2a4fc6d5.
And I modify the method to support Ruby 3 keyword arguments.